### PR TITLE
♻️ Use `hasOwn` instead of `Object.prototype.hasOwnProperty`.

### LINKED ISSUE #6548: Forbid invoke hasOwnProperty as method
`#hasOwnProperty` is almost exclusively used to test on objects you didn't create, and it breaks when that object is prototypeless. A much safer way is to use `Function#call` to invoke it with the object as the context:

```js
const hasOwnProperty = Object.prototype.hasOwnProperty;

if (hasOwnProperty.call(foreignObject, property) {
  // ...
}
```

Re: https://github.com/ampproject/amphtml/pull/6542

### PR DESCRIPTION
**Pull request by sparhami**
This is to make sure that the correct version of `hasOwnProperty` is called.

- Change type on `hasOwn` to allow null.
- Change various files to use `hasOwn`.
- Add conformance check preventing direct usage of `Object.prototype.hasOwnProperty`.

Fixes #6548

### CODE REVIEW

**Comment by sparhami** [comment]
This may be too big to get all the tests in the files touched to pass without flaking, but worth a try as a single change.

@jridgewell

File: src/utils/object.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ export function dict(opt_initial) {
  * Checks if the given key is a property in the map.
  *
  * @param {T}  obj a map like property.
- * @param {string}  key
+ * @param {?string}  key
```
Comment (jridgewell) [inline src/utils/object.js]: This will throw runtime errors. Was this just because CC was giving type errors?

Comment (jridgewell) [reply src/utils/object.js]: Oh man, I thought we were changing it to allow nullable `obj`, not the `key`. This is fine.

**Comment by sparhami** [comment]
@rsimha I see a test error on Travis that I don't see locally. Also, all the test cases do not appear to run. I tried creating a change that touches all the same files (but is a no-op) and see the same issue:

https://travis-ci.org/sparhami/amphtml/jobs/395680017
https://github.com/sparhami/amphtml/pull/1/files

Ali mentioned there is some difference between how the tests run locally vs on Travis. Any idea what needs to be done here?

/cc @aghassemi

**Comment by rsimha** [comment]
@sparhami You should see the same failures locally if you run `gulp test --local-changes`. Let me know if this isn't the case.

**Pull request closed without merging**